app/services/definitions.py

```diff
@@ -0,0 +1,543 @@
+        raise CloudApiException(
+            "An unexpected error occurred: could not publish schema."
+        )
+
+    result = credential_schema_from_acapy(result.sent.var_schema)
+    bound_logger.info("Successfully published and registered schema.")
+    return result
+
+
+async def get_schemas_tenant(
+    aries_controller: AcaPyClient,
+    schema_id: Optional[str],
+    schema_issuer_did: Optional[str],
+    schema_name: Optional[str],
+    schema_version: Optional[str],
+) -> List[CredentialSchema]:
+    """
+    Allows tenants to get all schemas created
+    """
+    bound_logger = logger.bind(
+        body={
+            "schema_id": schema_id,
+            "schema_issuer_did": schema_issuer_did,
+            "schema_name": schema_name,
+            "schema_version": schema_version,
+        }
+    )
+    bound_logger.debug("Fetching schemas from trust registry")
+
+    if not schema_id:  # client is not filtering by schema_id, fetch all
+        trust_registry_schemas = await get_trust_registry_schemas()
+    else:  # fetch specific id
+        trust_registry_schemas = [await get_trust_registry_schema_by_id(schema_id)]
+
+    schema_ids = [schema.id for schema in trust_registry_schemas]
+
+    bound_logger.debug("Getting schemas associated with fetched ids")
+    schemas = await schema_futures(schema_ids, aries_controller)
+
+    if schema_issuer_did:
+        schemas = [
+            schema for schema in schemas if schema.id.split(":")[0] == schema_issuer_did
+        ]
+    if schema_name:
+        schemas = [schema for schema in schemas if schema.name == schema_name]
+    if schema_version:
+        schemas = [schema for schema in schemas if schema.version == schema_version]
+
+    return schemas
+
+
+async def get_schemas_governance(
+    aries_controller: AcaPyClient,
+    schema_id: Optional[str],
+    schema_issuer_did: Optional[str],
+    schema_name: Optional[str],
+    schema_version: Optional[str],
+) -> List[CredentialSchema]:
+    """
+    Governance agents gets all schemas created by itself
+    """
+    bound_logger = logger.bind(
+        body={
+            "schema_id": schema_id,
+            "schema_issuer_did": schema_issuer_did,
+            "schema_name": schema_name,
+            "schema_version": schema_version,
+        }
+    )
+    # Get all created schema ids that match the filter
+    bound_logger.debug("Fetching created schemas")
+    response = await handle_acapy_call(
+        logger=bound_logger,
+        acapy_call=aries_controller.schema.get_created_schemas,
+        schema_id=schema_id,
+        schema_issuer_did=schema_issuer_did,
+        schema_name=schema_name,
+        schema_version=schema_version,
+    )
+
+    # Initiate retrieving all schemas
+    schema_ids = response.schema_ids or []
+
+    bound_logger.debug("Getting schemas associated with fetched ids")
+    schemas = await schema_futures(schema_ids, aries_controller)
+
+    return schemas
+
+
+async def schema_futures(
```

Comment:
I think futures refers to the awaitable before it's awaited. So this method can have a different name. Like get_schemas_by_id